airflow/models/dag.py

```diff
@@ -2404,6 +2404,20 @@ def __repr__(self):
     def timezone(self):
         return settings.TIMEZONE
 
+    @property
+    def timetable(self) -> Timetable:
+        interval = self.schedule_interval
+        if interval is None:
+            return NullTimetable()
+        if interval == "@once":
+            return OnceTimetable()
+        if isinstance(interval, (timedelta, relativedelta)):
+            return DeltaDataIntervalTimetable(interval)
+        if isinstance(interval, str):
+            return CronDataIntervalTimetable(interval, self.timezone)
+        type_name = type(interval).__name__
+        raise TypeError(f"{type_name} is not a valid DAG.schedule_interval.")
```

Comment:
Instead of duplicating the logic on `DagModel`, we should reuse `DAG.timetable` instead. But I’m not sure how this should be done.

Reply:
Yeah, thought about the same,
but I am also not sure, how to do it,
we should create a base table like DagBase which will have the common properties
including, which are there in both DagModel and DAG class
```python
    @property
    def safe_dag_id(self):
        return self.dag_id.replace('.', '__dot__')
    @cached_property
    def timetable(self) -> Timetable:
        interval = self.schedule_interval
        if interval is None:
            return NullTimetable()
        if interval == "@once":
            return OnceTimetable()
        if isinstance(interval, (timedelta, relativedelta)):
            return DeltaDataIntervalTimetable(interval)
        if isinstance(interval, str):
            return CronDataIntervalTimetable(interval, self.timezone)
        type_name = type(interval).__name__
        raise TypeError(f"{type_name} is not a valid DAG.schedule_interval.")
```